setup.py

```diff
@@ -43,7 +43,6 @@ def readme():
     'pytest-runner',
     'mock',
     'more-itertools~=5.0',  # We must pin this, because 6.0 requires Python 3.
-    'importlib-metadata',  # We need to explicitly install it to avoid incompatibilities with setuptools
```

Comment:
This was moved directly to the travis CI file. This was only an issue in python 3.7 and was still throwing the version compatibility error with this line here. My best guess is that the more recent version of `importlib-metadata` is needed to even begin to parse the `setup.py` file so installing it in the travis CI file seemed like the best solution.

Reply:
Does the removal impact local development? Shouldn't this be in both places (Tavis CI and setup.py)?

Reply:
The tests appeared to all still work for me locally with this change but I'll go through and re-create virtualenvs to make sure everything sets up properly

Reply:
A new virtualenv with python 3.7 (the version that broke in travis ci) worked locally for me but I can still add this back in if we feel it should be needed?

Reply:
I guess the error was caused for an incompatibility error with versions. Could we install the specific version you're using now but using setup.py? Also - this might require you update other deps versions or even install different importlib-metadata versions for specific python versions.

Reply:
Just added a change that puts this back in but only for the python versions that needed it, trying to pin the version for everything broke the lower python versions.